chains/solana/contracts/programs/ccip-router/src/instructions/v1/offramp.rs

```diff
@@ -668,12 +676,13 @@ pub fn validate_execution_report<'info>(
     Ok(())
 }
 
-fn hash(msg: &Any2SolanaRampMessage, on_ramp_address: &[u8]) -> [u8; 32] {
+fn hash(msg: &Any2SolanaRampMessage) -> [u8; 32] {
     use anchor_lang::solana_program::hash;
 
     // Calculate vectors size to ensure that the hash is unique
     let sender_size = [msg.sender.len() as u8];
-    let on_ramp_address_size = [on_ramp_address.len() as u8];
+    let trimmed_on_ramp_address = trim_leading_zeros(&msg.on_ramp_address);
+    let on_ramp_address_size = [trimmed_on_ramp_address.len() as u8];
```

Comment:
Related to my question regarding the type used to store the list of configured onramps: Any reason why not just hash the fixed 64 bytes we're allocating per address (with the necessary padding)? Just asking because it seems like changing those types might simplify the code a bit in a few places.

Reply:
I wasn't sure about this, I thought that this way will be more simple for the offchain code. But it makes sense we can force them to fit the 64bytes as well

Reply:
If we stick to the type being `Vec<u8>` with `max_len(128)`, I think what you did makes sense. My point is that if we change it to `[[u8; 64]; 2]`, then you can simplify code in a few different places, this being one of them by just always using 64 bytes.

This is all in the context of https://github.com/smartcontractkit/chainlink-ccip/pull/440#discussion_r1913757623